dandiapi/api/tests/test_version.py

```diff
@@ -359,6 +362,39 @@ def test_version_aggregate_assets_summary(draft_version_factory, draft_asset_fac
     assert version.metadata['assetsSummary']['schemaKey'] == 'AssetsSummary'
 
 
+@pytest.mark.django_db()
+def test_version_aggregate_assets_summary_metadata_modified(
+    draft_version_factory, draft_asset_factory
+):
+    version = draft_version_factory(status=Version.Status.VALID)
+    asset = draft_asset_factory(status=Asset.Status.VALID)
+    version.assets.add(asset)
+
+    # Modify the metadata passed to the function so that it's mismatched
+    version.metadata['foo'] = 'bar'
+    with pytest.raises(VersionMetadataConcurrentlyModified):
+        version_aggregate_assets_summary(version)
+
+
+@pytest.mark.django_db()
+def test_version_aggregate_assets_summary_task_retries(
+    draft_version_factory, draft_asset_factory, mocker, monkeypatch
+):
+    version = draft_version_factory(status=Version.Status.VALID)
```

Comment:
I'm not sure how I feel about this test. It looks it's mostly testing that Celery behaves correctly. If we're worried about configuration changing then I think that could be achieved with a test that just asserts `version_aggregate_assets_summary.autoretry_for == (VersionMetadataConcurrentlyModified,)`, but I would probably leave that out.

Reply:
I kind of agree, but it felt wrong having no test, since there's otherwise no way to verify that the task is actually retried :man_shrugging:. If you think it's more of a burden than it's worth, I can remove the test.

Reply:
I don't think what I'm saying is super authoritative, @mvandenburgh do you have an opinion?

Reply:
I guess I'll also just mention that the spirit of this test is more of a regression test than a behavioral test, if that makes a difference.

Reply:
> I kind of agree, but it felt wrong having no test, since there's otherwise no way to verify that the task is actually retried 🤷‍♂️. 

In principal I would argue that ensuring the task is retried is Celery's responsibility, and it's not something we need to test for ourselves. I think this is somewhat analogous to having a test that examines the RabbitMQ queue after calling `.delay()` to ensure the message is actually placed on the queue. As with other third party libraries, we trust Celery to do this correctly and treat it as a black box that "just works". In a similar manner, the fact that Celery will retry this task seems like a baseline assumption we should make, and the important thing to test is that `VersionMetadataConcurrentlyModified` is being raised correctly (which the other test does).

Reply:
Also, I realize that example is somewhat contrived because there have been real concerns raised about the stability of some of Celery's more advanced features/the guarantees it makes about them, but I think it's a slippery slope to start testing Celery features in our own test suite.